prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -565,6 +572,7 @@ class ActionHistory(
         result: String,
         header: WorkflowEngine.Header,
         reportEventService: IReportStreamEventService,
+        reportService: ReportService,
```

Comment:
Would this be better as a class member?

Reply:
It was really messy with how the action history gets instantiated to do it that way (also why I'm passed in the reportEventService that way originally).  I was bucketing this is in things to clean up if we refactor how some this works